### DISCUSSION TITLE
Is custom sorting when using custom data source supported ?

### DISCUSSION THREAD

**Discussion by gautamdsheth** [microsoft-search/pnp-modern-search#2707] [Q&A]
I am using the `pnp/modern-search-extensibility` to create a custom data source.
It is working fine and fetches the data correctly using the `getData` method.

I also specified the list of sortable fields in the `getSortableFields()` as mentioned in the docs here, https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_data_sources/#basedatasource-methods

What I am hoping for is that the fields would be available in the sort dropdown like this:

![image](https://user-images.githubusercontent.com/8832018/214261445-9fc5fc1a-8ed3-4757-9ef8-8064f07fab9b.png)

The fields specified in the `getSortableFields()` dont show up in the sorting dropdown.
Am I missing something ? Or doing something wrong ? Is this a supported scenario ? 

Is there a way for these custom fields to show up in the dropdown ?

**Comment by gretchunkim** [comment]
@gautamdsheth I'm also in the process of using pnp search extensibility to bring custom data source into pnp search results webpart due to sort feature limitation on externaItems as an entity type using Microsoft Graph data source.  

In order for even sort option to populate for in SharePoint as data source or some of the sort enabled feature in Microsoft Graph data source (such as non-externalItems as entity type), the connected data source property must be refinable. In Microsoft Graph data source, this is equivalent to marking source property to search, query, and retrieve in schema.  

I believe in custom data source (not Search & intelligence>Data sources, MS connectors) such as pnp search extensibility to bring custom data source, this option of defining schema isn't available.

**Reply by kasperbolarsen** [reply]
I remember Mikael Svenson saying that there isn't an API that can tell us which properties that are sortable/refinable.

**Reply by gretchunkim** [reply]
Yes, I do see that he has labels it as enhancement category for externalItems sorting feature. 

As for alternative, I’m all ears and eyes seeking other options.
